src/repositories/hardware_logs.rs

```diff
@@ -9,25 +9,48 @@ pub async fn fetch_foreign_matching_hardware<C: Context>(
     unique_id: &str,
     disk_id: &str,
 ) -> sqlx::Result<Vec<MatchingHardwareLog>> {
-    const QUERY: &str = const_str::concat!(
-        "SELECT hw.userid, u.username, u.privileges, ",
-        "hw.mac, hw.unique_id, hw.disk_id, ",
-        "SUM(hw.occurencies) AS occurencies, ",
-        "MAX(hw.activated) AS activated, ",
-        "MAX(hw.created_at) AS last_used ",
-        "FROM hw_user hw ",
-        "INNER JOIN users u ON hw.userid = u.id ",
-        "WHERE hw.userid != ? AND (hw.mac = ? OR hw.unique_id = ? OR hw.disk_id = ?) ",
-        "GROUP BY hw.mac, hw.unique_id, hw.disk_id, hw.userid ",
-        "ORDER BY hw.userid"
-    );
-    sqlx::query_as(QUERY)
-        .bind(user_id)
-        .bind(mac)
-        .bind(unique_id)
-        .bind(disk_id)
-        .fetch_all(ctx.db())
-        .await
+    // md5("runningunderwine"), osu! is running on wine
+    if mac == "b4ec3c4334a0249dae95c284ec5983df" {
+        // Only match by unique_id
+        // TODO: is matching by disk_id possible here?
+        const QUERY: &str = const_str::concat!(
+            "SELECT hw.userid, u.username, u.privileges, ",
+            "hw.mac, hw.unique_id, hw.disk_id, ",
+            "SUM(hw.occurencies) AS occurencies, ",
+            "MAX(hw.activated) AS activated, ",
+            "MAX(hw.created_at) AS last_used ",
+            "FROM hw_user hw ",
+            "INNER JOIN users u ON hw.userid = u.id ",
+            "WHERE hw.userid != ? AND hw.unique_id = ? ",
+            "GROUP BY hw.mac, hw.unique_id, hw.disk_id, hw.userid ",
+            "ORDER BY hw.userid"
+        );
+        sqlx::query_as(QUERY)
+            .bind(user_id)
+            .bind(unique_id)
+            .fetch_all(ctx.db())
+            .await
+    } else {
+        const QUERY: &str = const_str::concat!(
+            "SELECT hw.userid, u.username, u.privileges, ",
+            "hw.mac, hw.unique_id, hw.disk_id, ",
+            "SUM(hw.occurencies) AS occurencies, ",
+            "MAX(hw.activated) AS activated, ",
+            "MAX(hw.created_at) AS last_used ",
+            "FROM hw_user hw ",
+            "INNER JOIN users u ON hw.userid = u.id ",
+            "WHERE hw.userid != ? AND (hw.mac = ? OR hw.unique_id = ? OR hw.disk_id = ?) ",
```

Comment:
maybe worth leaving out restrictions for now & only logging?

Reply:
truth